File: front-end/portafolio/src/app/core/models/project.model.ts
export type ProjectStatus = 'draft' | 'published' | 'archived';

/**
 * Proyecto tal como lo expone la API. Home y cards usan el resumen
 * (`ProjectSummary`); el detalle/admin usan este modelo completo.
 */
export interface Project {
  id: string;
  title: string;
  description: string;
  short_description: string;
  image_url: string;
  github_url?: string;
  demo_url?: string;
  technologies: string[];
  highlights: string[];
  status: ProjectStatus;
  featured: boolean;
  order: number;
  created_at: string;
  updated_at: string;
}

/** Resumen para cards del Home y listados compactos. */
export interface ProjectSummary {
  id: string;
  title: string;
  short_description: string;
  image_url: string;
  technologies: string[];
  featured: boolean;
  created_at: string;
}

export interface ProjectListResponse {
  items: Project[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export interface ProjectForm {
  title: string;
  description: string;
  short_description: string;
  image_url: string;
  github_url?: string;
  demo_url?: string;
  technologies: string[];
  highlights: string[];
  featured?: boolean;
  order?: number;
  status?: ProjectStatus;
}

File: back-end/app/infrastructure/persistence/models/project_model.py
"""
Modelo SQLAlchemy para Project.
"""

from sqlalchemy import Column, String, Boolean, Integer, DateTime, Text, CHAR
from datetime import datetime
import uuid

from ..base import Base


class ProjectModel(Base):
    """Modelo de base de datos para proyectos."""
    
    __tablename__ = "projects"
    
    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String(200), nullable=False)
    description = Column(Text, nullable=False)
    short_description = Column(String(300))
    image_url = Column(String(500))
    github_url = Column(String(500))
    demo_url = Column(String(500))
    technologies = Column(Text)  # JSON string
    highlights = Column(Text)  # JSON string
    status = Column(String(20), default="draft")
    featured = Column(Boolean, default=False)
    order = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, onupdate=datetime.utcnow)
    
    def __repr__(self) -> str:
        return f"<Project {self.title}>"
